demos/text_detection_demo/src/text_recognition.cpp

```diff
@@ -53,3 +76,83 @@ std::string CTCGreedyDecoder(const std::vector<float> &data, const std::string&
     }
     return res;
 }
+
+std::string CTCBeamSearchDecoder(const std::vector<float> &data, const std::string& alphabet, char pad_symbol, double *conf, int bandwidth) {
+    std::string res = "";
+
+    const int num_classes = alphabet.length();
+
+    std::vector<BeamElement> curr;
+    std::vector<BeamElement> last;
+
+    last.push_back(BeamElement{"", 1.f, 0.f});
+
+    for (std::vector<float>::const_iterator it = data.begin(); it != data.end(); it += num_classes) {
+        curr.clear();
+
+        std::vector<float> prob = std::vector<float>(num_classes, 0.f);
+        softmax(it, it + num_classes, prob);
+
+        for (int candidate_num = 0; candidate_num < static_cast<int>(last.size()); candidate_num++) {
+            float prob_not_blank = 0.f;
+            auto candidate = last[candidate_num];
+            std::string candidate_sentence = candidate.sentence;
+            if (candidate_sentence != "") {
+                int n = static_cast<int>(candidate_sentence.back());
+                prob_not_blank = candidate.prob_blank * prob[n];
```

Comment:
2 @FengYen-Chang : I'm sorry, but this is a mistake.
The following code should be here:
```suggestion
                prob_not_blank = candidate.prob_not_blank * prob[n];
```
The reason is as follows: if we keep the `candidate_sentence` the same
(that is, if the new `BeamElement` added to the vector `curr` has the same field `sentence` as in the current `candidate`), 
then one of the following conditions should be fulfilled:
* either the last element in CTC sequence is the blank char -- in this case `prob_blank` field is increased
* or the last element in CTC sequence is the same as the the previous one AND the previous one is NOT a blank char -- in this case `prob_not_blank` field is increased, and for the new element 
    ```cpp
    prob_not_blank = candidate.prob_not_blank[index_of_last_symbol]
    ```
The latter case is handled in this line of the code.

Reply:
BTW, @Wovchena and me debugged this part of the code and found out that if we fix the mistake, the final confidence values of the results are better.

Reply:
Really thanks, it is my mistake, sorry. Thanks a lot.